esmvalcore/cmor/_fixes/cmip5/bcc_csm1_1.py

```diff
@@ -1,9 +1,65 @@
 """Fixes for bcc-csm1-1."""
+import iris
 import numpy as np
 from scipy.interpolate import InterpolatedUnivariateSpline
 from scipy.ndimage import map_coordinates
 
 from ..fix import Fix
+from ..shared import fix_bounds
+
+
+class Cl(Fix):
+    """Fixes for cl."""
+
+    def fix_metadata(self, cubes):
+        """Fix hybrid sigma pressure coordinate.
+
+        Parameters
+        ----------
+        cubes : iris.cube.CubeList
+            Input cubes which need to be fixed.
+
+        Returns
+        -------
+        iris.cube.CubeList
+
+        """
+        cl_cube = self.get_cube_from_list(cubes)
+
+        # Remove all existing aux_factories
+        for aux_factory in cl_cube.aux_factories:
+            cl_cube.remove_aux_factory(aux_factory)
+
+        # Fix bounds
+        coords_to_fix = ['b']
+        try:
+            cl_cube.coord(var_name='a')
+            coords_to_fix.append('a')
+        except iris.exceptions.CoordinateNotFoundError:
+            coords_to_fix.append('ap')
+        fix_bounds(cl_cube, cubes, coords_to_fix)
+
+        # Fix bounds for ap if only a is given in original file
+        ap_coord = cl_cube.coord(var_name='ap')
```

Comment:
How does this work? Shouldn't files only have either `a` or `ap`?

Reply:
Yes, the files have either a single `ap` coordinate or an `a` and `p0` coordinate. However, internally, `iris` converts the `a` and `p0` coordinate to `ap`, see

https://github.com/SciTools/iris/blob/06e9e2b5eae6e903e486ce12162c242d25ee4e5d/lib/iris/fileformats/netcdf.py#L677-L710

Reply:
Ok, got you. Then let me propose something.